src/js/components/deployment-view/build-log.tsx

```diff
@@ -0,0 +1,114 @@
+import * as classNames from 'classnames';
+import * as React from 'react';
+const Convert = require('ansi-to-html');
+
+import API from '../../api';
+import { Deployment } from '../../modules/deployments';
+
+import Spinner from '../common/spinner';
+
+const styles = require('./build-log.scss');
+
+const convert = new Convert({
+  fg: '#000',
+  bg: '#fff',
+  newline: true,
+  escapeXML: true,
+});
+
+interface PassedProps {
+  deployment: Deployment;
+}
+
+interface State {
+  log?: string;
+  error?: string;
+}
+
+type Props = PassedProps;
+
+interface ErrorResponse {
+  error: string;
+  details?: string | undefined;
+  unauthorized?: boolean | undefined;
+}
+
+function isError(response: any): response is ErrorResponse {
+  return (response as ErrorResponse).error !== undefined;
+}
+
+class BuildLog extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      log: undefined,
+      error: undefined,
+    };
+
+    this.updateLog = this.updateLog.bind(this);
+  }
+
+  private intervalID?: any;
+  private refreshCadenceInMs = 5000;
+
+  public componentWillMount() {
+    const { deployment } = this.props;
+
+    this.updateLog(deployment.id);
+    this.intervalID = setInterval(this.updateLog, this.refreshCadenceInMs, deployment.id);
+  }
+
+  private updateLog(deploymentId: string) {
+    API.Deployment.fetchBuildLog(deploymentId)
+      .then(result => {
+        if (isError(result)) {
+          if (this.intervalID) {
+            clearInterval(this.intervalID);
+          }
+
+          this.setState({ error: result.error });
+        } else {
+          this.setState({ log: result.response, error: undefined });
+        }
+      });
+  }
+
+  public componentWillReceiveProps(nextProps: Props) {
+    const { deployment } = this.props;
+
+    if (deployment.id !== nextProps.deployment.id) {
+      if (this.intervalID) {
+        clearInterval(this.intervalID);
+      }
+
+      this.updateLog(nextProps.deployment.id);
+      this.intervalID = setInterval(this.updateLog, this.refreshCadenceInMs, nextProps.deployment.id);
+    }
+  }
+
+  public render() {
+    const { log, error } = this.state;
+
+    if (!log && !error) {
+      return <Spinner />;
+    }
+
+    if (error) {
+      return (
+        <div className={classNames(styles.frame, styles.error)}>
+          Error: {error}
+        </div>
+      );
+    }
+
+    return (
+      <div
+        className={classNames(styles.frame, styles['build-log'])}
+        dangerouslySetInnerHTML={{ __html: convert.toHtml(log) }}
+      />
+    );
+  }
+}
+
+export default BuildLog;
```

Comment:
I get this warning in chrome console, we navigating away from the build log:

> warning.js?3efc:36 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the BuildLog component.

We should probably clear the interval on `componentWillUnmount` and maybe also check for `isMounted` before calling `setState`.